test/e2e/network_segmentation.go

```diff
@@ -599,68 +613,30 @@ var _ = Describe("Network Segmentation", func() {
 									"curl",
 									"--connect-timeout",
 									"2",
-									net.JoinHostPort(ip, fmt.Sprintf("%d", port)+"/hostname"),
+									net.JoinHostPort(ip, fmt.Sprintf("%d", httpServerPort)+"/hostname"),
 								)
-								Expect(err).NotTo(HaveOccurred())
-								if isRedPod {
-									Expect(strings.Contains(result, red)).To(BeTrue())
+
+								sameNetwork := isRedPod == redIPs[ip]
+								if !sameNetwork {
+									Expect(err).To(HaveOccurred(), "should isolate from different networks")
 								} else {
-									Expect(strings.Contains(result, blue)).To(BeTrue())
+									Expect(err).NotTo(HaveOccurred())
+									Expect(strings.Contains(result, expectedHostname)).To(BeTrue())
 								}
 							}
 						}
-
-						By("Deleting pods in network blue except " + fmt.Sprintf("%s-pod-%d", blue, numberOfPods-1))
-						for i := range numberOfPods - 1 {
-							err := cs.CoreV1().Pods(namespaceBlue).Delete(
-								context.Background(),
-								fmt.Sprintf("%s-pod-%d", blue, i),
-								metav1.DeleteOptions{},
-							)
-							Expect(err).NotTo(HaveOccurred())
-						}
-
-						podIP, err := podIPsForUserDefinedPrimaryNetwork(
-							cs,
-							namespaceBlue,
-							fmt.Sprintf("%s-pod-%d", blue, numberOfPods-1),
-							namespacedName(namespaceBlue, blue),
-							0,
-						)
-						Expect(err).NotTo(HaveOccurred())
-
-						By("Remaining blue pod cannot communicate with red networks overlapping CIDR")
-						for _, ip := range redIPs {
-							if podIP == ip {
-								//don't try with your own IP
-								continue
-							}
-							_, err := e2ekubectl.RunKubectl(
-								namespaceBlue,
-								"exec",
-								fmt.Sprintf("%s-pod-%d", blue, numberOfPods-1),
-								"--",
-								"curl",
-								"--connect-timeout",
-								"2",
-								net.JoinHostPort(ip, fmt.Sprintf("%d", port)),
-							)
-							Expect(err).To(MatchError(ContainSubstring("exit code 28")))
-						}
 					},
-					// can completely fill the L2 topology because it does not depend on the size of the clusters hostsubnet
 					Entry(
 						"with L2 primary UDN",
 						"layer2",
-						4,
-						"10.128.0.0/29",
-						"2014:100:200::0/125",
+						1, /* one pod per node */
+						userDefinedNetworkIPv4Subnet,
+						userDefinedNetworkIPv6Subnet,
 					),
-					// limit the number of pods to 10
 					Entry(
 						"with L3 primary UDN",
 						"layer3",
-						10,
+						2, /* two pods per node to share the subnet */
```

Comment:
hmm i am a bit confused here; isn't the goal to bring up two overlapping networks having pods with sameIPs? if you open up the subnet here to /16 doesn't that counter off the goal of this test?

Reply:
Right I was depending on the ip allocator dependency, let's go back to old values.